cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java

```diff
@@ -366,6 +369,20 @@ private boolean mergeHdr(Camera.Parameters params, Hdr oldHdr) {
         return false;
     }
 
+    @TargetApi(17)
+    private boolean mergePlaySound(boolean oldPlaySound) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            Camera.CameraInfo info = new Camera.CameraInfo();
+            Camera.getCameraInfo(mCameraId, info);
+            if (info.canDisableShutterSound && isCameraAvailable()) {
+                mCamera.enableShutterSound(mPlaySounds);
+                return true;
+            }
+        }
+        mPlaySounds = oldPlaySound;
```

Comment:
You should only revert it to the old value if (mPlaySounds == false). The true value should always be allowed, like...

if (apiLevel) { 
    ....
}
if (mPlaySounds) {
    return true; // always on
}
mPlaySounds = oldPlaySounds;
return false;